src/itdelatrisu/opsu/beatmap/BeatmapSet.java

```diff
@@ -43,24 +45,35 @@ public BeatmapSet(ArrayList<Beatmap> beatmaps) {
 	}
 
 	/**
-	 * Returns the number of elements.
+	 * Returns the number of search results or total number.
 	 */
-	public int size() { return beatmaps.size(); }
+	public int size() { return filteredBeatmaps.size() > 0 ? filteredBeatmaps.size() : beatmaps.size(); }
```

Comment:
At the moment how it is written is that filtered list starts off empty, and then maps are added to the filter. If the filter list is empty then it is assumed no filter is applied so therefore the unfiltered size should be returned, else other parts of the program will think there's no maps at all.  

I can change it so that instead filteredBeatmaps is a clone of beatmaps at first and then remove maps based on the criteria, that way it can always return filteredBeatmaps.size/get